### PR TITLE
HCK-7920: implemented variant subtype resolve logic on REing from ins…

### PR DESCRIPTION
**Pull request by WilhelmWesser**
<table>
<td>
  <a href="https://hackolade.atlassian.net/browse/HCK-7920" title="HCK-7920" target="_blank">Sub-taskHCK-7920</a>  Variant data type - RE from instance
  </td></table>
  <br />
 

## Content

Implemented logic for variant field subtype resolve in RE from instance

### CODE REVIEW

File: reverse_engineering/helpers/variantPropertiesSubtypeResolveHelper.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * @param {Primitive | object | Array<Primitive | object>} parsedRecord
+ * @returns {string}
+ */
+const getDocumentRecordType = parsedRecord => {
+	if (Array.isArray(parsedRecord)) {
+		return 'array';
+	}
+
+	if (typeof parsedRecord === 'object') {
+		return parsedRecord ? 'object' : null;
```
Comment (VitaliiBedletskyi) [inline reverse_engineering/helpers/variantPropertiesSubtypeResolveHelper.js:75]: According to the logic in `getMostFrequentValueInList` function and how the type name should be stored in a model - Should the null type be a string like "null"?

File: reverse_engineering/helpers/utils.js [hunk 2]
```diff
@@ -98,6 +98,42 @@ const getTemplateDocByJsonSchema = schema => {
...
+			}
+
+			return [mostFrequentItemValue, mostFrequentItemFrequency];
+		},
+		defaultMostFrequentEntry,
+	);
+
+	const [mostFrequentItemValue] = mostFrequentItemEntry;
+
+	return mostFrequentItemValue;
```
Comment (VitaliiBedletskyi) [inline reverse_engineering/helpers/utils.js:134]: A small suggestion is to make it more readable. If you agree, please apply it.
```suggestion
	let mostFrequentItem = '';
    let maxFrequency = 0;

    for (let [item, frequency] of itemsToFrequencyMap) {
        if (frequency > maxFrequency) {
            mostFrequentItem = item;
            maxFrequency = frequency;
        }
    }

    return mostFrequentItem;
```

File: reverse_engineering/helpers/utils.js [hunk 3]
```diff
@@ -98,6 +98,42 @@ const getTemplateDocByJsonSchema = schema => {
 	}, {});
 };
 
+/**
+ *
+ * @param {Array<number | string | boolean>} list
+ * @returns {number | string | boolean}
+ */
+const getMostFrequentValueInList = list => {
+	const itemsToFrequencyMap = {};
```
Comment (VitaliiBedletskyi) [inline reverse_engineering/helpers/utils.js:107]: it's one-to-one map (not many-to-one) so better to call it - `itemToFrequencyMap`

File: reverse_engineering/helpers/utils.js [hunk 4]
```diff
@@ -98,6 +98,42 @@ const getTemplateDocByJsonSchema = schema => {
 	}, {});
 };
 
+/**
+ *
+ * @param {Array<number | string | boolean>} list
+ * @returns {number | string | boolean}
+ */
+const getMostFrequentValueInList = list => {
```
Comment (VitaliiBedletskyi) [inline reverse_engineering/helpers/utils.js:106]: Based on the naming conventions used in the function, I propose naming it getMostFrequentItemInList.

**Review by VitaliiBedletskyi — CHANGES_REQUESTED** [review]

File: reverse_engineering/helpers/variantPropertiesSubtypeResolveHelper.js (hunk 1 above)
Comment (WilhelmWesser) [reply reverse_engineering/helpers/variantPropertiesSubtypeResolveHelper.js:75]: Yep, indeed

**Review by Vitalii4as — APPROVED** [review]

**Pull request merged**